internal/querynodev2/services.go

```diff
@@ -318,6 +324,9 @@ func (node *QueryNode) WatchDmChannels(ctx context.Context, req *querypb.WatchDm
 
 	err = loadGrowingSegments(ctx, delegator, req)
 	if err != nil {
+		// remove legacy growing
+		node.manager.Segment.RemoveBy(segments.WithChannel(channel.GetChannelName()),
+			segments.WithType(segments.SegmentTypeGrowing))
```

Comment:
should we use `defer` here to handle err return by `pipeline.ConcumeMsgStream` ?